src/main/java/org/terasology/genome/genomeMap/GeneIndexGenomeMap.java

```diff
@@ -81,6 +71,15 @@ public BreedingAlgorithm getPropertyBreedingAlgorithm(String propertyName) {
         return (T) definition.transformation.apply(genesForProperty.toString());
     }
 
+    /**
+     * Get the Property Definition Map for a Genome Map
+     *
+     * @return The property definition map
+     */
+    public Map<String, GenePropertyDefinition> getPropertyDefinitionMap() {
+        return propertyDefinitionMap;
```

Comment:
That works as well 😅  why did you decide go one step back with the abstractions and expose the map directly here? 
To be honest, I liked the idea of hiding whatever we are doing internally here, and just providing an access method to get a breeding algorithm by name.

Reply:
In addition, instead of returning the original map you may want to create a (immutable) copy like you do in `SimpleGenomeManager`

Reply:
Returning an immutable map is definitely needed, will change that. I went back on the abstraction and deleted the getBreedingAlgorithm method as the propertyDefinitionMap contains more fields that are used in conjunction with the breeding algorithm and therefore when a particular field of the map is accessed, it returns a GenePropertyDefinition object which contains the breeding algorithm. Since this is already accessed, it did not feel necessary to have another method to get the same thing (since we already accessed it as a side product of accessing a value of the propertyDefinitionMap)